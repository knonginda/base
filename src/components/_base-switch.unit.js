/* eslint-disable no-undef */
import BaseSwitch from './_base-switch'

describe('@components/_base-checkbox', () => {
  it('renders its content', () => {
    const wrapper = shallowMount(BaseSwitch, {
      propsData: {
        onLabel: 'Yes',
        offLabel: 'No',
      },
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('default value changes', () => {
    const wrapper = shallowMount(BaseSwitch, {
      propsData: {
        onLabel: 'Yes',
        offLabel: 'No',
      },
    })
    const checkbox = wrapper.find('input[type="checkbox"]')
    checkbox.setChecked(true)
    expect(wrapper.emitted('input', true)).toEqual([[true]])
    checkbox.setChecked(false)
    expect(wrapper.emitted('input', false)).toEqual([[true], [false]])
  })
})
