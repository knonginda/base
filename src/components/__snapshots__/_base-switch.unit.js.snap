// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/_base-checkbox renders its content 1`] = `
<div class="medium switch"><input id="checkbox-id-1" type="checkbox">
  <div class="switchInner"><label for="checkbox-id-1"></label> <span class="switchText">
      No
    </span></div>
</div>
`;
